Doc/library/functions.rst

```diff
@@ -514,44 +514,40 @@ are always available.  They are listed here in alphabetical order.
 
 .. function:: eval(expression, /, globals=None, locals=None)
 
-   The arguments are a string and optional globals and locals.  If provided,
+   This function dynamically evaluates Python expressions. The first argument can
+   be a string or a :ref:`code object <datamodel-codeobjects>`. The optional
+   arguments specify the global and local namespaces respectively. If provided,
    *globals* must be a dictionary.  If provided, *locals* can be any mapping
    object.
 
-   The *expression* argument is parsed and evaluated as a Python expression
-   (technically speaking, a condition list) using the *globals* and *locals*
-   dictionaries as global and local namespace.  If the *globals* dictionary is
-   present and does not contain a value for the key ``__builtins__``, a
-   reference to the dictionary of the built-in module :mod:`builtins` is
-   inserted under that key before *expression* is parsed.  That way you can
-   control what builtins are available to the executed code by inserting your
-   own ``__builtins__`` dictionary into *globals* before passing it to
-   :func:`eval`.  If the *locals* dictionary is omitted it defaults to the
-   *globals* dictionary.  If both dictionaries are omitted, the expression is
-   executed with the *globals* and *locals* in the environment where
-   :func:`eval` is called.  Note, *eval()* does not have access to the
-   :term:`nested scopes <nested scope>` (non-locals) in the enclosing
-   environment.
+   When the *expression* argument is a string, it is parsed and evaluated as a
+   Python expression (see :ref:`expression-input`). The leading and trailing spaces,
+   tabs, and newlines are stripped.
+
+   The evaluation is performed in the environment specified by the arguments
+   *globals* and *locals*.  If both are omitted, by default it uses the
+   environment where :func:`eval` is called.  If only the *globals* argument is
+   provided, the local namespace defaults to *globals*.
+
+   Before evaluation, the special key ``"__builtins__"`` is searched for in the
```

Comment:
Not too certain about quoting the key, and perhaps it could be non-literal and just "__builtins__" instead, but I'll defer judgement here.